src/test/java/spoon/test/model/TypeTest.java

```diff
@@ -8,21 +8,28 @@
 import org.junit.Test;
 
 import spoon.reflect.declaration.CtType;
-import spoon.reflect.declaration.CtType;
 import spoon.reflect.reference.CtExecutableReference;
 
 public class TypeTest {
 
 	@Test
-	public void test() throws Exception {
+	public void testGetAllExecutables() throws Exception {
 		CtType<?> type = build("spoon.test.model", "Foo");
 		assertEquals(1, type.getDeclaredFields().size());
-		assertEquals(2, ((CtType) type).getMethods().size());
-		assertEquals(3, type.getDeclaredExecutables().size());
+		assertEquals(3, type.getMethods().size());
+		assertEquals(4, type.getDeclaredExecutables().size());
 		assertEquals(2, type.getAllFields().size());
 		
-		// we have 4  methods + one explicit constructor + two implicit constructors for Bar and Baz
+		// we have 4  methods + one explicit constructor + two implicit
+		// constructors for Bar, Baz and Baz.Inner
 		Collection<CtExecutableReference<?>> allExecutables = type.getAllExecutables();
-		assertEquals(7, allExecutables.size());
+		assertEquals(8, allExecutables.size());
+	}
+
+	@Test
+	public void testGetUsedTypes() throws Exception {
+		CtType<?> type = build("spoon.test.model", "Foo");
+		assertEquals(3, type.getUsedTypes(true).size());
```

Comment:
can you write a stronger assert, e.g. `assertTrue(type.getUsedTypes(true).contains(Baz.class)`